bioptim/optimization/optimal_control_program.py

```diff
@@ -743,7 +743,326 @@ def load(file_path: str) -> list:
+                else:
+                    name = parameter.penalty_list.name
+            return name
+
+        def print_console(ocp):
+            for phase_idx in range(ocp.n_phases):
+                print(f"PHASE {phase_idx}")
+                print(f"**********")
+                print(f"PARAMETERS: ")
+                print("")
+                for parameter in ocp.nlp[phase_idx].parameters:
+                    initial_guess, min_bound, max_bound = scaling_parameter(parameter)
+                    objective_name = get_parameter_function_name(parameter)
+                    print(f"Name: {parameter.name}")
+                    print(f"Size: {parameter.size}")
+                    print(f"Initial_guess: {initial_guess}")
+                    print(f"Min_bound: {min_bound}")
+                    print(f"Max_bound: {max_bound}")
+                    print(f"Objectives: {objective_name}")
+                    print("")
+                print("")
+                print(f"**********")
+                print(f"MODEL: {ocp.original_values['biorbd_model'][phase_idx]}")
+                print(f"PHASE DURATION: {round(ocp.nlp[phase_idx].t_initial_guess, 2)} s")
+                print(f"SHOOTING NODES : {ocp.nlp[phase_idx].ns}")
+                print(f"DYNAMICS: {ocp.nlp[phase_idx].dynamics_type.type.name}")
+                print(f"ODE: {ocp.nlp[phase_idx].ode_solver.rk_integrator.__name__}")
+                print(f"**********")
+                print("")
+
+                mayer_objectives = mayer_to_str(ocp.nlp[phase_idx].J)
+                lagrange_objectives = lagrange_to_str(ocp.nlp[phase_idx].J)[1]
+                print(f"*** Lagrange: ")
+                for name in lagrange_objectives:
+                    print(name)
+                print("")
+                for node_idx in range(ocp.nlp[phase_idx].ns):
+                    print(f"NODE {node_idx}")
+                    print(f"*** Mayer: ")
+                    for mayer in mayer_objectives:
+                        if mayer[0] == node_idx:
+                            print(mayer[1])
+                    for i in range(ocp.nlp[phase_idx].g.__len__()):
+                        if ocp.nlp[phase_idx].g[i][0]['node_index'] == node_idx:
+                            print(f"*** Constraint {i}: {ocp.nlp[phase_idx].g[phase_idx][i]['constraint'].name}")
+                    print("")
+
+        def draw_graph(ocp):
+
+            # Initialize Digraph using graphviz
+            from graphviz import Digraph
+            G = Digraph('ocp_graph', node_attr={'shape': 'plaintext'})
+
+            # We draw nlp clusters composed of different nodes
+            def draw_nlp_cluster(phase_idx: int):
+
+                def draw_parameter_node(param_idx: int, parameter: Parameter):
+                    initial_guess, min_bound, max_bound = scaling_parameter(parameter)
+                    node_str = f"<u><b>{parameter.name}</b></u><br/>"
```

Comment:
nested of nested of nested function is probably a design flaw symptom. Please make sure this function cannot be moved to draw_graph or even in print_ocp_structure (which would already make them private for the functions

Reply:
Done